model/configuration.py

```diff
@@ -382,14 +368,12 @@ def for_collection_and_purpose(cls, _db, collection, purpose):
     @classmethod
     def lookup(cls, _db, protocol, goal, library=None):
 
-        integrations = (
-            _db.query(cls)
-            .outerjoin(cls.libraries)
```

Comment:
No need for this join in the common case where we are not filtering on libraries at all.